spring-integration-core/src/test/java/org/springframework/integration/handler/MethodInvokingMessageProcessorTests.java

```diff
@@ -583,26 +586,30 @@ public void testPerformanceSpelVersusInvocable() throws Exception {
 		StopWatch stopWatch = new StopWatch("SpEL vs Invocable Performance");
 
 		stopWatch.start("SpEL");
-		for (int i = 0; i < 10000; i++) {
+		int count = 10_000;
+		for (int i = 0; i < count; i++) {
 			processor.processMessage(message);
 		}
 		stopWatch.stop();
 
 		processor = new MethodInvokingMessageProcessor(service, method);
 
 		stopWatch.start("Invocable");
-		for (int i = 0; i < 10000; i++) {
+		for (int i = 0; i < count; i++) {
 			processor.processMessage(message);
 		}
 		stopWatch.stop();
 
+		// Update the parser configuration compiler mode
+		SpelParserConfiguration config = TestUtils.getPropertyValue(processor,
+				"delegate.handlerMethod.EXPRESSION_PARSER.configuration", SpelParserConfiguration.class);
+		new DirectFieldAccessor(config).setPropertyValue("compilerMode", SpelCompilerMode.IMMEDIATE);
```

Comment:
I found one of the performance bottleneck - `InvocableHandlerMethod` : 
```
private Object[] getMethodArgumentValues(Message<?> message, Object... providedArgs) throws Exception {
	MethodParameter[] parameters = getMethodParameters();
        ...
	for (int i = 0; i < parameters.length; i++) {
		MethodParameter parameter = parameters[i];
	...
		GenericTypeResolver.resolveParameterType(parameter, getBean().getClass());

```
The `GenericTypeResolver.resolveParameterType()` always does the same for the same `MethodParameter` object for nothing on each method invocation!

I did some local hacking moving such a initialization to the ctor and increase performance for `1_000_000` from:
```
02284  085%  SpEL
00339  013%  Invocable
00056  002%  Compiled SpEL
```
to
```
02243  087%  SpEL
00262  010%  Invocable
00071  003%  Compiled SpEL
```
Yeah... not a big deal but still.
I guess the rest of time we spend on the reflection method invocation instead of direct call from compiled SpEL...

Merging and opening JIRA against SF at least to see what guys think